backend/coreapp/models/github.py

```diff
@@ -215,17 +215,18 @@ def pull(self) -> None:
     def get_dir(self) -> Path:
         return Path(settings.LOCAL_FILE_DIR) / "repos" / str(self.id)
 
-    def details(self, access_token: str) -> Repository:
-        cache_key = f"github_repo_details:{self.id}"
-        cached = cache.get(cache_key)
-
-        if cached:
-            return cached
-
-        details = Github(access_token).get_repo(f"{self.owner}/{self.repo}")
+    def get_sha(self) -> str:
+        repo_dir = self.get_dir()
+        if not repo_dir.exists():
+            raise RuntimeError("Repo directory does not exist.")
```

Comment:
This check should be done in `get_dir` I think; it should exist always otherwise it implies the db is inconsistent with the filesystem